api/src/State/ResendActivationProcessor.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace App\State;
+
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\State\ProcessorInterface;
+use App\DTO\UserActivation;
+use App\Entity\User;
+use App\Repository\UserRepository;
+use App\Security\ReCaptcha\ReCaptchaWrapper;
+use App\Service\MailService;
+use App\Util\IdGenerator;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\NonUniqueResultException;
+use Doctrine\ORM\NoResultException;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+
+/**
+ * @implements ProcessorInterface<UserActivation,null>
+ */
+readonly class ResendActivationProcessor implements ProcessorInterface {
+    public function __construct(
+        private ReCaptchaWrapper $reCaptcha,
+        private EntityManagerInterface $em,
+        private UserRepository $userRepository,
+        private MailService $mailService,
+    ) {}
+
+    /**
+     * @param UserActivation $data
+     *
+     * @throws NoResultException
+     * @throws NonUniqueResultException
+     */
+    public function process($data, Operation $operation, array $uriVariables = [], array $context = []): null {
+        $resp = $this->reCaptcha->verify($data->recaptchaToken);
+        if (!$resp->isSuccess()) {
+            throw new HttpException(422, 'ReCaptcha failed');
+        }
+
+        $user = $this->userRepository->loadUserByIdentifier($data->email);
+
+        if (null == $user) {
+            return null;
+        }
+
+        if (!in_array($user->state, [User::STATE_NONREGISTERED, User::STATE_REGISTERED])) {
```

Comment:
Why is `resend_activation` also allowed for users in `NONREGISTERED` status?

Reply:
tried to use the same if statement as in https://github.com/ecamp/ecamp3/blob/c4181e17d2642ed9fc0106d14335fab4ef5a7208/api/src/Security/OAuth/HitobitoAuthenticator.php#L79

But could set User::state to null, thus the null value fell out of the array of possible stati.

I chose the defensive way that a user can still activate himself, even when our system broke somehow and left the state in NONREGISTERED.

Reply:
`NONREGISTERED` means that the user has been inserted by someone else (e.g. through an invitation). If we collect more data during the registration process in the future (or take more data from OAuth), this PR can be used to bypass this.

I think that resend-activation should only be allowed for users with the status `REGISTERED`.

Reply:
done